plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTask.java

```diff
@@ -206,12 +212,17 @@ public void performAction(IncrementalTaskInputs inputs) throws Exception {
 			if (getCacheFile().exists()) {
 				LastApply lastApply = SerializableMisc.fromFile(LastApply.class, getCacheFile());
 				for (File file : lastApply.changedFiles) {
-					if (!outOfDate.contains(file) && file.exists() && Iterables.contains(target, file) && shouldInclude.test(file)) {
+					if (shouldInclude.test(file) && !outOfDate.contains(file) && file.exists() && Iterables.contains(target, file)) {
 						outOfDate.add(file);
 					}
 				}
 			}
 
+			if (outOfDate.isEmpty()) {
+				// no work to do
+				return;
+			}
```

Comment:
The try block should start after this.

Reply:
sorry, I misunderstood before.
updated again